GitUI/UserControls/RevisionGrid/FilterInfo.cs

```diff
@@ -303,6 +309,77 @@ public ArgumentString GetRevisionFilter()
                 filter.Add($"--max-count={CommitsLimit}");
             }
 
+            if (!ShowMergeCommits)
+            {
+                filter.Add("--no-merges");
+            }
+
+            if (ShowOnlyFirstParent)
+            {
+                filter.Add("--first-parent");
+            }
+
+            // Listed in Git help as history simplification, but is a revision filter
+            if (ShowSimplifyByDecoration)
+            {
+                filter.Add("--simplify-by-decoration");
+            }
+
+            if (ByAuthor && !string.IsNullOrWhiteSpace(Author))
+            {
+                filter.Add($"--author=\"{Author}\"");
+            }
+
+            if (ByCommitter && !string.IsNullOrWhiteSpace(Committer))
+            {
+                filter.Add($"--committer=\"{Committer}\"");
+            }
+
+            if (IgnoreCase && (ByAuthor || ByCommitter || ByMessage || ByDiffContent))
+            {
+                filter.Add("--regexp-ignore-case");
+            }
+
+            if (ByDateFrom)
+            {
+                filter.Add($"--since=\"{DateFrom:yyyy-MM-dd hh:mm:ss}\"");
+            }
+
+            if (ByDateTo)
+            {
+                filter.Add($"--until=\"{DateTo:yyyy-MM-dd hh:mm:ss}\"");
+            }
+
+            if (ByDiffContent && !string.IsNullOrWhiteSpace(DiffContent))
+            {
+                filter.Add($"-G\"{DiffContent}\"");
+            }
+
+            if (ByMessage && !string.IsNullOrWhiteSpace(Message))
+            {
+                if (Message.StartsWithAny(_messageAsGitOptions))
```

Comment:
`Message` must be trimmed to account for any whitespaces

Reply:
this time I prefer to not change, this is a special handling....

Reply:
I think it's well worth adding several test cases with various content of `Message` to document the intended/expected behaviours.

Reply:
> " --exclude bla" won't be treated as the filter though

Yes, my intention. Special handling to have special rules.
I can change this though.

Reply:
My concern is that users may find it unintuitive. E.g., a user may copy-paste the "--not" or "--exclude" filter from somewhere, and that string may very well start with a whitespace (space, tab, non-breakable space, etc.).